aten/src/ATen/native/quantized/cpu/qconcat.cpp

```diff
@@ -0,0 +1,63 @@
+#include <ATen/ATen.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+
+#include <algorithm>
+#include <vector>
+
+namespace at { namespace native {
+namespace {
+
+bool is_valid_quantization_type(Tensor t) {
+  const auto qtype = t.qscheme();
+  return (qtype == kPerTensorAffine) ||
+         (qtype == kPerTensorSymmetric);
+}
+
+template <bool ReLUFused = false>
+class QCat final : public torch::OperatorKernel {
+ public:
+  Tensor operator()(const std::vector<Tensor>& qxs, int64_t axis,
+                    c10::optional<double> scale,
+                    c10::optional<int64_t> zero_point) {
+    const auto x_dtype = qxs[0].scalar_type();
+    double _scale = scale ? *scale : qxs[0].q_scale();
+    int64_t _zero_point = zero_point ? *zero_point : qxs[0].q_zero_point();
+
+    std::vector<Tensor> xs;
+    xs.reserve(qxs.size());
+    for (const auto& qx: qxs) {
+      TORCH_CHECK(x_dtype == qx.scalar_type(),
+                   "All dtypes must be the same.");
+      TORCH_CHECK(is_valid_quantization_type(qx),
```

Comment:
should we check all the quantization schemes match? just like we checked whether all the scalar_types match here?
I know in this implementation since we are doing dequantize it would not matter, but I assume if we do this in fbgemm, we will require everything matches.

Reply:
Should I implement the check, or should I leave a TODO?

Reply:
@zafartahirov please don't resolve conversation that still needs response

Reply:
I think to keep the behavior consistent with future implementation of fbgemm we should just add the check now

Reply:
The reason I resolved it is that I implemented the request -- just need to sent the PR